paddle/memory/CMakeLists.txt

```diff
@@ -1,14 +1,16 @@
 add_subdirectory(detail)
 
 cc_library(memory SRCS memory.cc)
-cc_library(memcpy SRCS memcpy.cc)
+cc_library(memcpy SRCS memcpy.cc DEPS device_context)
```

Comment:
I don't think `GPUPlaceGuard` and `CPUPlaceGuard ` are necessary here. We don't need to roll back the previous device id. Because we still need to set device id when we want to use it.